runners/samza/src/main/java/org/apache/beam/runners/samza/runtime/SamzaDoFnRunners.java

```diff
@@ -439,6 +445,25 @@ public void finishBundle() {
       }
     }
 
+    private void closeBundle() throws Exception {
+      long bundleProcessingTimeout = pipelineOptions.getBundleProcessingTimeout();
+      if (bundleProcessingTimeout < 0) {
+        // RemoteBundle close blocks until all results are received
+        remoteBundle.close();
+      } else {
+        CompletableFuture<Void> future =
+            CompletableFuture.runAsync(
+                () -> {
+                  try {
+                    remoteBundle.close();
+                  } catch (Exception e) {
+                    throw new RuntimeException(e);
```

Comment:
Why not throw the same exception instead of wrapping it up in `RuntimeException`?

Reply:
The `close()` method throws the checked exception `Exception`, we have to handle the checked exception explicitly in the Java lamda expression.